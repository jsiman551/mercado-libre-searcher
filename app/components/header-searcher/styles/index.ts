import { styled } from "styled-components";
import { 
    desktop, 
    tablet,
} from "@/constants/styles";

/************Styles For Header Section**************/
/* Header where searcher is */
export const Header = styled.header`
    background: #FEE600;
    display: flex; 
    align-items: center;
    padding: .6rem 1rem .6rem 1rem;
    @media ${tablet} {
        padding: .6rem 2rem .6rem 2rem;
    }
    @media ${desktop} {
        padding: .6rem 0 .6rem 0;
        justify-content: center;
    }
`;

/* Header elements container (form) */
export const ElementsContainer = styled.form`
    display: flex;
    align-items: center;
    width: 100%;
    @media ${desktop} {
        max-width: 1000px;
    }
`;

/* input to search for items */
export const SearchInput = styled.input`
    height: 40px;
    color: #333;
    padding: 0px 60px 0px 15px;
    border-radius: 4px;
    border-color: rgba(0,0,0,0);
    margin-left: 1.5rem;
    width: 100%;
    &:focus {
        outline: none;
    }
`;

/* button to submit search info */
export const SearchButton = styled.button`
    height: 43px;
    min-width: 50px;
    margin-left: -50px;
    background: #EEEEEE;
    border: 0;
    border-top-right-radius: 4px;
    border-bottom-right-radius: 4px;
    padding-top: 4px;
    cursor: pointer;
`;
