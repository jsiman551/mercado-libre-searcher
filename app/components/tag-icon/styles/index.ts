import { styled } from "styled-components";

/*****Tag Icon Styles******/
export const TagIconContainer = styled.span<{ $bgColor: string }>`
  background: ${(props) => props.$bgColor};
  border-radius: 50%;
  display: flex;
  align-items: center;
  justify-content: center;
  width: 20px;
  height: 20px;
  margin-left: 8px;
`;
