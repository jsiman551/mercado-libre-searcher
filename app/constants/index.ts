/* API URL */
export const ApiUrl: String = 'https://api.mercadolibre.com/sites/MLA/search'
