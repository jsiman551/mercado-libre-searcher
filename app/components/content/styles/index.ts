import styled from "styled-components";
import { tablet, desktop } from "@/constants/styles";

/* Content Container */
export const ContentContainer = styled.div`
  display: flex;
  justify-content: center;
  margin: 1rem auto 0 auto;
  padding: 0 1rem 0 1rem;
  @media ${tablet} {
    padding: 0 2rem 0 2rem;
  }
  @media ${desktop} {
    padding: 0;
    max-width: 1050px;
  }
`;

/* only appears on long resolutions */
export const OnlyLongResolutions = styled.div`
  display: none;
  @media ${tablet} {
    display: contents;
  }
`;
