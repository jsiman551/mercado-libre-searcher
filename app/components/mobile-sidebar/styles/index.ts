import { styled } from "styled-components";
import { tablet } from "@/constants/styles";

/****Sidebar For Filters in Mobile Resolution*****/
/* overlay element */
export const SidebarOverlay = styled.div<{ $isOpen: boolean; }>`
    position: fixed;
    height: 100%;
    left: 0;
    top: 0;
    background: white;
    z-index: 10;
    box-shadow: 2px 0px 2px rgba(0,0,0,.25);
    transition: transform .5s;
    transform: ${props => props.$isOpen ? "translate(0)" : "translate(-246px)"};
    @media ${tablet} {
        display: none;
    }
`;

/* close button styles */
export const SidebarCloseButton = styled.button`
    float: right;
    background: transparent;
    border: 1px solid rgba(0,0,0,.25);
    border-radius: 4px;
    font-size: .875em;
    margin: 0.4rem 0.4rem 0 0
`;