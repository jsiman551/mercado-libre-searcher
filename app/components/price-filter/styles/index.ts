import { styled } from "styled-components";
import { desktop, tablet } from "@/constants/styles";

/****************Price Filter Styles*********************/
/* Price Filter Container */
export const Container = styled.div`
    font-size: 1em;
    box-sizing: border-box;
    padding-right: .5rem;
    border-bottom: 1px solid rgba(0,0,0,.25);
    width: 80%;
    margin: 2rem 0 0 1rem;
    @media ${tablet} {
        border: 0;
        margin-left: 0;
        width: 45%;
        padding-top: 3rem;
    }
    @media ${desktop} {
        width: 30%;
    }
`;

/* Price Label */
export const Label = styled.h3`
    color: #333;
    font-size: 1em;
    font-weight: 600;
    line-height: 1.25;
    margin-bottom: 8px;
`;

/* Price List Container */
export const ListContainer = styled.ul`
    padding: 0;
    margin-top: 0;
`;

/* Price List Element */
export const RangeElement = styled.li`
    border-collapse: collapse;
    box-sizing: border-box;
    list-style: none;
    font-size: .875em;
    margin: 0 0 6px;
    cursor: pointer;
`;

/* Price Link */
export const RangeLink = styled.a`
    border-collapse: collapse;
    font-size: .875em;
    box-sizing: border-box;
    text-decoration: none;
    display: flex;
    flex-wrap: wrap;
    width: fit-content;
    &:hover span {
        color: #3483fa;
    }
`;

/* Price Link Name */
export const RangeLinkName = styled.span`
    border-collapse: collapse;
    box-sizing: border-box;
    color: #666;
    float: left;
    font-size: .875em;
    margin-left: 0;
`;

/* Price Link Quantity */
export const RangeLinkQuantity = styled.span`
    border-collapse: collapse;
    box-sizing: border-box;
    color: #999;
    float: left;
    font-size: .875em;
    margin-left: 6px;
`;

/* Price Filter Form */
export const RangeForm = styled.form`
    font-size: 1em;
    line-height: 1.35;
    list-style: none;
    box-sizing: border-box;
    display: flex;
    flex-direction: row;
`;

/* Price Input Container */
export const InputContainer = styled.div`
    font-size: 1em;
    align-items: center;
    display: flex;
    &:first-child {
        &:after {
            color: #999;
            content: "â€”";
            float: left;
            margin: 0 5px;
            padding: 7px 0;
            font-size: 0.75em;
        }
    }
`;

/* Price Filter Input */
export const RangeInput = styled.input`
    box-sizing: border-box;
    border: 0;
    color: rgba(0,0,0,.9);
    margin: 0;
    padding: .8125em .75em;
    background-color: #fafafa;
    border-radius: .375em;
    font-size: 0.813em;
    height: 25px;
    width: 84px;
    box-shadow: 0 0 0 1px rgba(0,0,0,.25);
    &:focus {
        outline: 0;
        box-shadow: 0 0 0 2px #fff, 0 0 0 3px #2968C8;
    }
`;

/* Price Filter Button */
export const RangeButton = styled.button<{ $active: boolean; }>`
    align-items: center;
    border: none;
    border-radius: 50%;
    cursor: pointer;
    display: flex;
    justify-content: center;
    height: 24px;
    margin-left: 8px;
    width: 24px;
    background-color: ${props => props.$active ? "#3483fa" : "#e0e0e0"};;
`;
