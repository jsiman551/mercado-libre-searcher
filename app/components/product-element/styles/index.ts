import { styled } from 'styled-components'
import { desktop, mobile, tablet } from '@/constants/styles'

/************Styles For Articles List Section**************/
/* product list container */
export const ProductListContainer = styled.div`
  background: white;
  padding: 1rem;
  border-radius: 4px;
  margin-top: 1rem;
  margin-bottom: 3rem;
`

/* product element container */
export const Container = styled.div`
  display: flex;
  justify-content: space-between;
  border-bottom: 2px solid #f4f4f4;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  flex-direction: column;
  &:last-child {
    border-bottom: unset;
  }
  @media ${tablet} {
    flex-direction: row;
  }
`

/* product details container */
export const DetailsContainer = styled.div`
  display: flex;
  flex-direction: column;
  @media ${tablet} {
    flex-direction: row;
    align-items: center;
    width: 80%;
  }
`

/* product image Container */
export const ImageContainer = styled.div`
  text-align: center;
  @media ${tablet} {
    text-align: left;
  }
`

/* product image */
export const ProductImage = styled.img`
  border-radius: 4px;
  object-fit: contain;
`

/* product text details container */
export const Details = styled.div`
  margin-left: 1rem;
`

/* tag for product price text */
export const PriceText = styled.p`
  color: #333;
  font-size: 1em;
  margin-bottom: 0;
  display: flex;
  align-items: center;
  @media ${desktop} {
    font-size: 1.25em;
  }
`

/* tag for product Title text */
export const TitleText = styled.h2`
  margin-bottom: 0;
  font-size: 1em;
  margin-top: 14px;
  font-weight: 400;
  @media ${desktop} {
    font-size: 1.25em;
  }
`

/* tag for product dues text */
export const DuesText = styled.p`
  margin-bottom: 0;
  color: #427d2c;
  font-size: 0.875em;
  @media ${desktop} {
    font-size: 1em;
  }
`

/* tag for product location text */
export const LocationText = styled.p`
  color: #8b8b8b;
  font-size: 0.75em;
  margin-top: 2rem;
  @media ${mobile} {
    margin-top: 1rem;
    text-align: center;
  }
  @media ${tablet} {
    margin-top: 2rem;
    text-align: left;
    width: 20%;
  }
`
