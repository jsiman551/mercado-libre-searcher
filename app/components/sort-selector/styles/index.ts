import styled from "styled-components";
import { tablet, desktop } from "@/constants/styles";

/**************Sort Selector Styles*****************/
/* Selector Container */
export const Container = styled.div`
  display: flex;
  margin-left: auto;
  margin-right: auto;
  justify-content: space-between;
  align-items: center;
  @media ${tablet} {
    justify-content: end;
  }
`;

export const FlexBox = styled.div`
  display: flex;
  justify-content: end;
`;

/* Selector Label */
export const Label = styled.div`
  margin-right: 0.4rem;
  font-weight: 500;
  font-size: 0.875em;
`;

/* Selector Text */
export const OptionText = styled.div`
  display: flex;
  align-items: center;
  font-size: 0.875em;
  cursor: pointer;
  &:hover {
    color: #3483fa;
  }
`;

/* Selector Arrow Icon */
export const ArrowIcon = styled.img<{ $active: boolean }>`
  transform: ${(props) => (props.$active ? "rotate(180deg)" : "rotate(0deg)")};
  transition: transform 0.2s ease-out;
  margin-top: 3px;
  margin-left: 2px;
`;

/* Selector Elements Container */
export const ElementContainer = styled.ul`
  border-radius: 0.375em;
  position: absolute;
  top: 101px;
  background-color: #fff;
  padding: 0;
  box-shadow: 0 1px 2px 0 rgba(0, 0, 0, 0.12);
  font-size: 0.875em;
  width: 134px;
  @media ${desktop} {
    padding-right: 1rem;
  }
`;

/* Selector Elements Option */
export const ElementOption = styled.li<{ $active: boolean }>`
  border-collapse: collapse;
  color: ${(props) => (props.$active ? "#3483fa" : "#666")};
  box-sizing: border-box;
  cursor: pointer;
  align-items: center;
  display: flex;
  height: 41px;
  border-bottom: 1px solid #d8d8d8;
  padding: 1rem;
  font-weight: ${(props) => (props.$active ? "500" : "400")};
  &:before {
    background: ${(props) => (props.$active ? "#3483fa" : "")};
    content: "";
    position: absolute;
    width: 5px;
    height: 41px;
    left: 0;
  }
  &:hover {
    &:after {
      background: ${(props) => (props.$active ? "#3483fa" : "#9cc4db")};
      content: "";
      left: 0;
      position: absolute;
      width: 5px;
      height: 41px;
    }
    background-color: rgba(0, 0, 0, 0.04);
    transition: background-color 0.1s ease-out;
  }
  &:first-child {
    border-top-left-radius: 0.375em;
    border-top-right-radius: 0.375em;
    &:after {
      border-top-left-radius: 0.375em;
    }
    &:before {
      border-top-left-radius: 0.375em;
    }
  }
  &:last-child {
    border: 0;
    &:after {
      border-bottom-left-radius: 0.375em;
    }
    &:before {
      border-bottom-left-radius: 0.375em;
    }
  }
`;

/****Sidebar For Filters in Mobile Resolution*****/
export const SidebarButton = styled.button`
  display: flex;
  background-color: white;
  border-radius: 4px;
  border: 1px solid;
  cursor: pointer;
  @media ${tablet} {
    display: none;
  }
`;
