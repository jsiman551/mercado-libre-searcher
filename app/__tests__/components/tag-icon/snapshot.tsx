import { render } from '@testing-library/react'
import TagIcon from '@/components/tag-icon'
import { styleTheme } from '@/constants/styles'
import { ThemeProvider } from 'styled-components'

/* for Free Shipping Icon */
const mockProps = {
  imgUrl: '/icons/truck-icon.svg',
  bgColor: 'rgb(46, 200, 102)',
  altDescription: 'free shipping',
}

it('renders TagIcon component with expected content for Free Shipping Icon', () => {
  const { container } = render(
    <ThemeProvider theme={styleTheme}>
      <TagIcon {...mockProps} />
    </ThemeProvider>,
  )
  expect(container).toMatchSnapshot()
})
