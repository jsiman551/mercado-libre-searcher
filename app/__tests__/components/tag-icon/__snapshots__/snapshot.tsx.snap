// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders TagIcon component with expected content for Free Shipping Icon 1`] = `
<div>
  <span
    class="sc-beyTiQ euMBhX"
    data-testid="tagIconContainer"
  >
    <img
      alt="free shipping"
      data-nimg="1"
      decoding="async"
      height="12"
      loading="lazy"
      src="/icons/truck-icon.svg"
      style="color: transparent;"
      width="12"
    />
  </span>
</div>
`;
